src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/Setup/hooks/useDefaultColumnConfig.js

```diff
@@ -54,12 +54,8 @@ const getProgramStageMainConfig = (programStage): Array<MetadataColumnConfig> =>
 
 const getEventsMetaDataConfig = (programStage): Array<MetadataColumnConfig> => {
     const stageForm = programStage.stageForm;
-    const sections = programStage.stageForm?.sections ? [...stageForm.sections?.values()] : [];
-
-    return sections.reduce((acc, section) => {
-        const dataElements = [...section.elements.values()];
-        return [...acc, ...getDataValuesMetaDataConfig(dataElements)];
-    }, []);
+    const dataElements = programStage.stageForm ? [...stageForm?.getElements()] : [];
```

Comment:
Refactoring this since flow the stage form might not only be DataElements any more. Could refactor more in the future as the stageForm has some very form specific metadata connected to data entry. Should probably build this metadata on demand from indexedDB as well. @JoakimSM

Reply:
Great! Yeah, indexedDB makes sense in future